pgfinder/matching.py

```diff
@@ -105,43 +104,22 @@ def builder(name, mass, mult_num: int):
                 theo_mw.append(Decimal(mw) + donor_mw + Decimal("-18.0106"))
                 theo_struct.append(acceptor + "-" + donor + "|" + str(mult_num))
 
-    # Calls builder subfunction with different arguements based on multimer type selected
-
-    # Calculates multimers based on peptide bond through side chain
-
+    # Call builder subfunction with different arguements based on multimer type selected
+    # and calculate multimers based on peptide bond through side chain
+    # if multimer_type is "peptide":
     if multimer_type == 0:
-        builder("gm-AEJA", Decimal("941.4075"), 2)
-        builder("gm-AEJ", Decimal("870.3704"), 2)
-        builder("gm-AEJ-gm-AEJ", Decimal("1722.7302"), 3)
-        builder("gm-AEJ-gm-AEJA", Decimal("1793.7673"), 3)
-        builder("gm-AEJA-gm-AEJA", Decimal("1864.8044"), 3)
-
-    # Calculates multimers based on glycosidic bond through dissachrides & peptide bonds through side chains
+        multimer = MULTIMERS["peptide"]
+    # elif multimer_type is "gycosidic":
     elif multimer_type == 1:
-        builder("gm-AE", Decimal("698.2858"), 2)
-        builder("gm-AEJA", Decimal("941.4075"), 2)
-        builder("gm-AEJ", Decimal("870.3704"), 2)
-        builder("gm-AEJ-gm-AEJ", Decimal("1722.7302"), 3)
-        builder("gm-AEJ-gm-AEJA", Decimal("1793.7673"), 3)
-        builder("gm-AEJA-gm-AEJA", Decimal("1864.8044"), 3)
-
-        builder("gm-AEJA_(Glyco)", Decimal("939.3919"), 2)
-        builder("gm-AEJ_(Glyco)", Decimal("868.3548"), 2)
-        builder("gm-AEJ-gm-AEJ_(Glyco)", Decimal("1720.7146"), 3)
-        builder("gm-AEJ-gm-AEJA_(Glyco)", Decimal("1791.7517"), 3)
-        builder("gm-AEJA-gm-AEJA_(Glyco)", Decimal("1862.7888"), 3)
-
-    # Calculates multimers based on Lactyl peptides (peptide bond via side chain but no dissachrides on muropeptides)
+        multimer = MULTIMERS["gycosidic"]
```

Comment:
Really like the use of research area terminology in the code rather than type 1, type 2 - this should help researchers understand the code better.